--- a/drivers/net/ethernet/airoha/Makefile
+++ b/drivers/net/ethernet/airoha/Makefile
@@ -1,9 +1,16 @@
 # SPDX-License-Identifier: GPL-2.0-only
 #
-# Airoha for the Mediatek SoCs built-in ethernet macs
+# Airoha and EcoNet frame-engine Ethernet drivers
 #
 
 obj-$(CONFIG_NET_AIROHA) += airoha-eth.o
 airoha-eth-y := airoha_eth.o airoha_ppe.o
-airoha-eth-$(CONFIG_DEBUG_FS) += airoha_ppe_debugfs.o
+airoha-eth-$(CONFIG_NET_AIROHA_PPE_DEBUGFS) += airoha_ppe_debugfs.o
+airoha-eth-$(CONFIG_NET_AIROHA_SOC_WED) += airoha_wed.o airoha_wed_debugfs.o
+airoha-eth-$(CONFIG_NET_AIROHA_WHNAT) += airoha_whnat.o
+
 obj-$(CONFIG_NET_AIROHA_NPU) += airoha_npu.o
+
+# Airoha/EcoNet shared GPON + EPON MAC complex.
+obj-$(CONFIG_AIROHA_XPON_V1) += airoha-xpon.o
+airoha-xpon-y := airoha_xpon.o airoha_ploam.o airoha_gpon_omci.o
