--- a/drivers/net/pcs/Kconfig
+++ b/drivers/net/pcs/Kconfig
@@ -4,6 +4,13 @@
 #
 
 menu "PCS device drivers"
+
+config FWNODE_PCS
+	tristate
+	depends on (ACPI || OF)
+	depends on PHYLINK
+	help
+		Firmware node PCS accessors
 
 config PCS_XPCS
 	tristate "Synopsys DesignWare Ethernet XPCS"
@@ -20,10 +27,24 @@
 
 config PCS_MTK_LYNXI
 	tristate
+	select PHY_COMMON_PROPS
 	select REGMAP
 	help
 	  This module provides helpers to phylink for managing the LynxI PCS
 	  which is part of MediaTek's SoC and Ethernet switch ICs.
+
+config PCS_MTK_USXGMII
+	tristate "MediaTek USXGMII PCS"
+	select FWNODE_PCS
+	select PCS_MTK_LYNXI
+	select PHY_COMMON_PROPS
+	select PHYLINK
+	imply PHY_MTK_PEXTP
+	help
+	  This module provides a driver for MediaTek's USXGMII PCS supporting
+	  10GBase-R, 5GBase-R and USXGMII interface modes.
+	  1000Base-X, 2500Base-X and Cisco SGMII are supported on the same
+	  differential pairs via an embedded LynxI PCS.
 
 config PCS_RZN1_MIIC
 	tristate "Renesas RZ/N1, RZ/N2H, RZ/T2H MII converter"
@@ -34,4 +55,6 @@
 	  Renesas RZ/N1, RZ/N2H, and RZ/T2H SoCs. This PCS converts MII to
 	  RMII/RGMII, or can be set in pass-through mode for MII.
 
+source "drivers/net/pcs/airoha/Kconfig"
+
 endmenu
